### PR TITLE
docs: add native param types page

### PR DESCRIPTION
**Pull request by sarahschwartz**
This PR adds a new page in the Types section of the docs to explain some common contract input and output types for native Sway blockchain types.

### CODE REVIEW

File: apps/docs-snippets/src/guide/types/contract-types.test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+describe(__filename, () => {
+  let contract: Contract;
+
+  beforeAll(async () => {
+    contract = await createAndDeployContractFromProject(DocSnippetProjectsEnum.INPUT_OUTPUT_TYPES);
+  });
+
+  it('should successfully call a function with an Address type input and output parameters', async () => {
+    // #region address-input
```
Comment (petertonysmith94) [inline apps/docs-snippets/src/guide/types/contract-types.test.ts:18]:
```suggestion
    // #region address-input
    // #import { Address };
```

**Review by petertonysmith94 — COMMENTED** [review]
Not sure whether the imports were intentional left out or not - so please disregard if so.

File: apps/docs-snippets/src/guide/types/contract-types.test.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    // #region contract-id-output
+    const contractIdOutput = callResponse.value;
+    const contractIdFromOutput: string = contractIdOutput.value;
+    // #endregion contract-id-output
+
+    expect(contractIdFromOutput).toEqual(contractId);
+  });
+
+  it('should successfully call a function with a Identity type input and output parameters', async () => {
+    // #region identity-address-input
```
Comment (petertonysmith94) [inline apps/docs-snippets/src/guide/types/contract-types.test.ts:48]:
```suggestion
    // #region identity-address-input
    // #import { Address };
```

File: apps/docs-snippets/src/guide/types/contract-types.test.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    const address = Address.fromRandom();
+    const addressInput = { value: address.toB256() };
+    const addressIdentityInput = { Address: addressInput };
+    // #endregion identity-address-input
+    const callResponse1 = await contract.functions.identity(addressIdentityInput).simulate();
+
+    // #region identity-address-output
```
Comment (petertonysmith94) [inline apps/docs-snippets/src/guide/types/contract-types.test.ts:55]:
```suggestion
    // #region identity-address-output
    // #import { Address };
```

File: apps/docs-snippets/src/guide/types/contract-types.test.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+  it('should successfully call a function with an Address type input and output parameters', async () => {
+    // #region address-input
+    // #import { Address };
+    const address = Address.fromRandom();
+    const addressInput = { value: address.toB256() };
+    // #endregion address-input
+    const callResponse = await contract.functions.address(addressInput).simulate();
+
+    // #region address-output
```
Comment (petertonysmith94) [inline apps/docs-snippets/src/guide/types/contract-types.test.ts:25]:
```suggestion
    // #region address-output
    // #import { Address };
```

File: apps/docs/.vitepress/config.ts [hunk 5]
```diff
@@ -164,6 +164,10 @@ export default defineConfig({
                 text: 'Date Time',
                 link: '/guide/types/date-time',
               },
+              {
```
Comment (petertonysmith94) [inline apps/docs/.vitepress/config.ts:167]: Is there an order to this list?
Could this be higher? Seems like valuable knowledge.

**Review by petertonysmith94 — COMMENTED** [review]
That's all - looks good 🥇

File: apps/docs/src/guide/types/native-parameters.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+### `AddressInput`
+
+To pass an `Address` as an input parameter to a contract, you can define the input as shown below:
+
+<<< @/../../docs-snippets/src/guide/types/contract-types.test.ts#address-input
+
+### `AddressOutput`
+
+For a contract call that returns an `Address` type in Sway, you can convert to an `Address` type in `fuels` as shown below:
```
Comment (arboleya) [inline apps/docs/src/guide/types/native-parameters.md]:
```suggestion
For a contract call that returns an `Address` type in Sway, you can convert it to an `Address` type in `fuels` as shown below:
```

File: apps/docs/src/guide/types/native-parameters.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+### `ContractIdInput`
+
+To pass a `ContractId` as an input parameter to a contract, you can define the input as shown below:
+
+<<< @/../../docs-snippets/src/guide/types/contract-types.test.ts#contract-id-input
+
+### `ContractIdOutput`
+
+For a contract call that returns a `ContractId` type in Sway, you can convert to an `string` as shown below:
```
Comment (arboleya) [inline apps/docs/src/guide/types/native-parameters.md]:
```suggestion
For a contract call that returns a `ContractId` type in Sway, you can convert it to a `string` as shown below:
```

File: apps/docs/src/guide/types/native-parameters.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+<<< @/../../docs-snippets/src/guide/types/contract-types.test.ts#identity-address-input
+
+For a contract:
+
+<<< @/../../docs-snippets/src/guide/types/contract-types.test.ts#identity-contract-input
+
+### `IdentityOutput`
+
+For a contract call that returns an `Identity` type in Sway, you can convert to an `Address` or `string` as shown below:
```
Comment (arboleya) [inline apps/docs/src/guide/types/native-parameters.md]:
```suggestion
For a contract call that returns an `Identity` type in Sway, you can convert it to an `Address` or `string` as shown below:
```

File: apps/docs/src/guide/types/native-parameters.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+### `AssetIdInput`
+
+To pass an `AssetId` as an input parameter to a contract, you can define the input as shown below:
+
+<<< @/../../docs-snippets/src/guide/types/contract-types.test.ts#asset-id-input
+
+### `AssetIdOutput`
+
+For a contract call that returns an `AssetId` type in Sway, you can convert to a `string` as shown below:
```
Comment (arboleya) [inline apps/docs/src/guide/types/native-parameters.md]:
```suggestion
For a contract call that returns an `AssetId` type in Sway, you can convert it to a `string` as shown below:
```

**Review by arboleya — APPROVED** [review]
Only nits.

Great work @sarahschwartz!

Thank you. 🙌

**Review by petertonysmith94 — APPROVED** [review]

**Pull request merged**